src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStep.java

```diff
@@ -50,6 +54,14 @@ public String getScript() {
         return new BourneShellScript(script);
     }
 
+    @Override public StepExecution start(StepContext context) throws Exception {
+        String path = context.get(EnvVars.class).get("PATH");
+        if (path != null && path.contains("$PATH")) {
+            context.get(TaskListener.class).getLogger().println("Warning: JENKINS-41339 probably bogus PATH=" + path + "; perhaps you meant to use ‘PATH+EXTRA=/something/bin’?");
```

Comment:
Not necessary for `EnvironmentVariablesNodeProperty`, which is now fixed, but perhaps for people using `withEnv` incorrectly etc.